union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
```

Comment:
I think that returning `Map(nil)` is not the ideal behavior here for a few reasons:

* I believe it'll cause a panic when you try to perform an FS operation, because `Map` doesn't make an empty map if its argument is nil. That is solvable, but...
* At a conceptual level, what is the union of 0 filesystems? I would argue it is a meaningless concept and should be `nil`. Here you *create* a new real filesystem and return it.

So I think it would be better to mention in the documentation that if no filesystems are passed to `Union`, it returns `nil` (or panics)--and implement it that way, too.